src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -244,16 +280,35 @@ void RadioSpinel<InterfaceType>::Init(bool aResetRadio, bool aSkipRcpCompatibili
     SuccessOrDie(error);
 }
 
+template <typename InterfaceType>
+void RadioSpinel<InterfaceType>::SetCallbacks(const struct RadioSpinelCallbacks &aCallbacks)
+{
+#if OPENTHREAD_CONFIG_DIAG_ENABLE
+    assert(aCallbacks.mDiagReceiveDone != nullptr);
+    assert(aCallbacks.mDiagTransmitDone != nullptr);
+#endif
+    assert(aCallbacks.mEnergyScanDone != nullptr);
+    assert(aCallbacks.mReceiveDone != nullptr);
+    assert(aCallbacks.mTransmitDone != nullptr);
+    assert(aCallbacks.mTxStarted != nullptr);
+
+    mCallbacks = aCallbacks;
```

Comment:
Is it purposely skipped to check the `mSwitchoverDone` callback in `SetCallbacks()` API? Also I don’t see where is switchover callback set from?

Reply:
Yes, this is intentional. Some hosts may want to be notified of the interface switch and take some actions and some may not care. Some platforms may operate in parallel on all interfaces and there would be no interface switching - there is no point implementing a handler in such case. For ot-daemon, I do not see a use for this.
Note: There is no host using multiple interfaces yet, so trying to make this api as generic as possible. We may need to change/extend this interface in the future (e.g. we may want to add target interface id for the message, for now I wanted to keep LAST_STATUS message as short as possible).